### PR TITLE
DAOSGCP-167 test: Remove hard-coded /mnt/daos from general_utils.py

### PR DESCRIPTION
**Pull request by shimizukko**
Remove hard-coded /mnt/daos from check_pool_files() and check_for_pool() and move them to test_utils_pool.py

Update tests to pass in scm_mount.

Update tags.

Skip-unit-tests: true
Skip-fault-injection-test: true
Test-tag: MultiServerCreateDeleteTest DynamicServerPool test_destroy_single test_destroy_multi test_destroy_single_loop test_destroy_multi_loop test_destroy_invalid_uuid test_destroy_invalid_label test_destroy_wrong_group test_destroy_connected test_destroy_with_containers
Required-githooks: true

### Before requesting gatekeeper:

* [ ] Two review approvals and any prior change requests have been resolved.
* [ ] Testing is complete and all tests passed or there is a reason documented in the PR why it should be force landed and forced-landing tag is set.
* [ ] `Features:` (or `Test-tag*`) commit pragma was used or there is a reason documented that there are no appropriate tags for this PR.
* [ ] Commit messages follows the guidelines outlined [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments).
* [ ] Any tests skipped by the ticket being addressed have been run and passed in the PR.

### Gatekeeper:

* [ ] You are the appropriate gatekeeper to be landing the patch.
* [ ] The PR has 2 reviews by people familiar with the code, including appropriate watchers.
* [ ] Githooks were used. If not, request that user install them and check copyright dates.
* [ ] Checkpatch issues are resolved.  Pay particular attention to ones that will show up on future PRs.
* [ ] All builds have passed.  Check non-required builds for any new compiler warnings.
* [ ] Sufficient testing is done. Check feature pragmas and test tags and that tests skipped for the ticket are run and now pass with the changes.
* [ ] If applicable, the PR has addressed any potential version compatibility issues.
* [ ] Check the target branch.   If it is master branch, should the PR go to a feature branch?  If it is a release branch, does it have merge approval in the JIRA ticket.
* [ ] Extra checks if forced landing is requested
  * [ ] Review comments are sufficiently resolved, particularly by prior reviewers that requested changes.
  * [ ] No new NLT or valgrind warnings.  Check the classic view.
  * [ ] Quick-build or Quick-functional is not used.
* [ ] Fix the commit message upon landing. Check the standard [here](https://daosio.atlassian.net/wiki/spaces/DC/pages/11133911069/Commit+Comments). Edit it to create a single commit. If necessary, ask submitter for a new summary.

### CODE REVIEW

File: src/tests/ftest/util/test_utils_pool.py [hunk 1]
```diff
@@ -1422,3 +1424,49 @@ def measure_rebuild_time(self, operation, interval=1):
...
+
+    def check_pool_files(self, log, hosts, uuid, scm_mount):
+        """Check if pool files exist on the specified list of hosts.
+
+        Args:
+            log (logging): logging object used to display messages
+            hosts (NodeSet): list of hosts
+            uuid (str): uuid file name to look for in /mnt/daos.
+            scm_mount (str): SCM mount point such as "/mnt/daos". From test, it can be
+                obtained as self.server_managers[0].get_config_value("scm_mount")
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:1436]: Does this function actually need to accept `uuid` or could that just come from `self.uuid.lower()`?
Also, I think `log` can just be `self.log` now, right?

File: src/tests/ftest/util/test_utils_pool.py [hunk 2]
```diff
@@ -1422,3 +1424,49 @@ def measure_rebuild_time(self, operation, interval=1):
...
+        Returns:
+            bool: True if pool folder exists, False otherwise
+
+        """
+        pool_dir = f"{scm_mount}/{uuid}"
+        result = check_file_exists(host, pool_dir, directory=True, sudo=True)
+        if result[0]:
+            print("{} exists on {}".format(pool_dir, host))
+        else:
+            print("{} does not exist on {}".format(pool_dir, host))
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:1471]: We should replace `print` with `self.log`

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/test_utils_pool.py (hunk 1 above)
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_pool.py]: I checked the usage of `check_pool_files()` and noticed that `test_destroy_invalid_uuid()` in destroy.py updates the pool's `uuid` member to invalid one, then passes in valid UUID to this method, so we need this parameter.

I removed `log` parameter and just use `self.log`

File: src/tests/ftest/util/test_utils_pool.py (hunk 2 above)
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_pool.py]: Done

File: src/tests/ftest/util/test_utils_pool.py [hunk 3]
```diff
@@ -1422,3 +1424,48 @@ def measure_rebuild_time(self, operation, interval=1):
...
+                obtained as self.server_managers[0].get_config_value("scm_mount")
+
+        Returns:
+            bool: True if pool folder exists, False otherwise
+
+        """
+        pool_dir = f"{scm_mount}/{uuid}"
+        result = check_file_exists(host, pool_dir, directory=True, sudo=True)
+        if result[0]:
+            self.log.info("{} exists on {}".format(pool_dir, host))
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:1468]: Pylint
> logging-format-interpolation, Use lazy % formatting in logging functions

```suggestion
            self.log.info("%s exists on %s", pool_dir, host)
```

File: src/tests/ftest/util/test_utils_pool.py [hunk 4]
```diff
@@ -1422,3 +1424,48 @@ def measure_rebuild_time(self, operation, interval=1):
...
+        else:
+            self.log.info("{} does not exist on {}".format(pool_dir, host))
```
Comment (daltonbohning) [inline src/tests/ftest/util/test_utils_pool.py:1470]:
```suggestion
            self.log.info("%s does not exist on %s", pool_dir, host)
```

**Review by daltonbohning — CHANGES_REQUESTED** [review]

File: src/tests/ftest/util/test_utils_pool.py (hunk 1 above)
Comment (daltonbohning) [reply src/tests/ftest/util/test_utils_pool.py]: I see. I have an idea how to rework that but for now I think it's out of scope for this PR

File: src/tests/ftest/util/test_utils_pool.py (hunk 3 above)
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_pool.py]: Done

File: src/tests/ftest/util/test_utils_pool.py (hunk 4 above)
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_pool.py]: Done

**Review by daltonbohning — APPROVED** [review]

**Review by phender — DISMISSED** [review]

File: src/tests/ftest/util/test_utils_pool.py [hunk 5]
```diff
@@ -1422,3 +1424,48 @@ def measure_rebuild_time(self, operation, interval=1):
...
+        self.log.info("Checking for pool data on %s", hosts)
+        pool_files = [uuid, "superblock"]
+        for filename in [f"{scm_mount}/{item}" for item in pool_files]:
+            result = check_file_exists(hosts, filename, sudo=True)
+            if not result[0]:
+                self.log.error("%s: %s not found", result[1], filename)
+                status = False
+        return status
+
+    def check_for_pool(self, host, uuid, scm_mount):
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_pool.py]: Done

**Comment by shimizukko** [comment]
The failure of scrubber/target_auto_eviction.py isn't related to this PR. Created a ticket: [DAOS-14585](https://daosio.atlassian.net/browse/DAOS-14585)

**Review by daltonbohning — APPROVED** [review]

**Review by daltonbohning — APPROVED** [review]

**Review by dinghwah — APPROVED** [review]

File: src/tests/ftest/util/test_utils_pool.py [hunk 6]
```diff
@@ -1421,3 +1423,48 @@ def measure_rebuild_time(self, operation, interval=1):
...
+        self.log.info("Checking for pool data on %s", hosts)
+        pool_files = [uuid, "superblock"]
+        for filename in [f"{scm_mount}/{item}" for item in pool_files]:
+            result = check_file_exists(hosts, filename, sudo=True)
+            if not result[0]:
+                self.log.error("%s: %s not found", result[1], filename)
+                status = False
+        return status
+
+    def verify_uuid_directory(self, host, uuid, scm_mount):
```
[the remark this answers is not in the document]
Comment (shimizukko) [reply src/tests/ftest/util/test_utils_pool.py]: Removed. Thanks for pointing it out.

**Pull request merged**